testcases/PowerNVDump.py

```diff
@@ -1324,6 +1330,262 @@ def runTest(self):
+        test_binaries to root directory and compiles and
+        creates "watchdog-countdown" tool directory,using
+        this can trigger the watchdog events.
+        '''
+        filename = "watchdog-countdown.c"
+        self.cv_HOST.copy_test_file_to_host(filename, dstdir="/root")
+        base_file = os.path.splitext(filename)[0]
+        if not self.cv_HOST.host_run_command(f"make {base_file}"):
+            self.fail("make file command failed,Please check the file")
+
+    def module_load_with_parameters(self, i_module, timeout, action):
+        '''
+        This function will load the module using modprobe
+        with timeout and action parameters and valiadates
+        module load.
+        :params
+            i_module: watchdog Module name "pseries_wdt".
+            timeout: timeout value while loading module
+            action: type of reset.
+                    Action_0 : halt the LPAR
+                    Action_1 : reboot the LPAR
+                    Action_2 : collect dump and reboot LPAR
+        :rtype int
+        '''
+        try:
+            cmd = f"modprobe {i_module} timeout={timeout} action={action}"
+            self.cv_HOST.host_run_command(cmd)
+            self.script_timeout = timeout
+            self.script_action = action
+            return self.script_timeout, self.script_action
+
+        except CommandFailed as c:
+            l_msg = "Error in loading the module %s, modprobe failed: %s" % (
+                i_module, str(c))
+            raise OpTestError(l_msg)
+
+    def validate_timeout_and_action(self):
+        '''
+        Funtion to validate timeout and action of watchdog module.
+        '''
+        self.get_watchdog_timeout_value()
+        self.get_watchdog_action_value()
+
+        if (self.system_timeout_value == self.script_timeout) and (self.watchdog_mode == self.script_action):
+            return True
+        else:
+            self.fail("Timeout and action values of watchdog module"
+                      "are incorrect,Please check logs")
+
+    def get_watchdog_timeout_value(self, i_module):
+        '''
+        Funtion to get timeout value of a watchdog_module
+        that set by script.
+
+        :params i_module: watchdog module i.e "pseries_wdt"
+        :rtype int
+        '''
+        cmd = f"cat /sys/module/{i_module}/parameters/timeout"
+        output = self.cv_HOST.host_run_command(cmd)
+        self.system_timeout_value = int(output[0])
+        return self.system_timeout_value
+
+    def get_watchdog_action_value(self, i_module):
+        '''
+        Funtion to get action value of a watchdog_module
+        that set by script.
+
+        :params i_module: watchdog module i.e "pseries_wdt"
+        :rtype int
+        '''
+        cmd = f"cat /sys/module/{i_module}/parameters/action"
+        output = self.cv_HOST.host_run_command(cmd)
+        self.watchdog_mode = int(output[0])
+        return self.watchdog_mode
```

Comment:
is it mode or some value it is returning. keep meaning full name

Reply:
Changed to self.watchdog_action_mode
 and also doc string is avialable.